### PR TITLE
Options to resize & rotate symbols

### PR DESCRIPTION
**Pull request by XiaoMigros**
Resolves: #17116
Resolves: #18462
Resolves: https://github.com/musescore/MuseScore/issues/15522

This PR introduces the options to scale and rotate symbol elements. These options can be managed via the properties tab, where it's now possible to change the symbol's font as well.

- [x] I signed the [CLA](https://musescore.org/en/cla)
- [x] The title of the PR describes the problem it addresses
- [x] Each commit's message describes its purpose and effects, and references the issue it resolves
- [ ] If changes are extensive, there is a sequence of easily reviewable commits
- [x] The code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] There are no unnecessary changes
- [x] The code compiles and runs on my machine, preferably after each commit individually
- [ ] I created a unit test or vtest to verify the changes I made (if applicable)

### CODE REVIEW

**Review by oktophonie — APPROVED** [review]

File: src/engraving/dom/symbol.cpp [hunk 1]
```diff
@@ -84,6 +96,16 @@ PropertyValue Symbol::getProperty(Pid propertyId) const
     switch (propertyId) {
     case Pid::SYMBOL:
         return PropertyValue::fromValue(m_sym);
+    case Pid::SCORE_FONT:
+        if (m_scoreFont) {
+            return PropertyValue::fromValue(String::fromStdString(m_scoreFont->name()));
+        } else {
+            return PropertyValue::fromValue(nullptr);
```
Comment (mike-spa) [inline src/engraving/dom/symbol.cpp:103]: Maybe using an empty string rather than a nullptr looks better (and same below)?

**Review by mike-spa — APPROVED** [review]

**Comment by bkunda** [comment]
@XiaoMigros I'd love to be able to review this, but the macOS build appears to have failed.

**Comment by XiaoMigros** [comment]
Very sorry about that, @cbjeukendrup has told me that I'm not the only encountering this issue so I hope it can be resolved soon..

**Comment by Jojo-Schmitz** [comment]
That failure looks similar to that from #21360

**Comment by Jojo-Schmitz** [comment]
As other more recent PRs succeed, maybe a rebase helps?

**Comment by XiaoMigros** [comment]
Seems to have worked!

**Review by bkunda — CHANGES_REQUESTED** [review]
Nice work @XiaoMigros :-) 

1. I wonder whether we really need the font dropdown? @oktophonie do any of the other music fonts currently support the symbols in the master palette?
2. We need an icon for the button in properties (visible when multiple elements are selected). 

Screenshot 2024-02-07 at 2 53 20 pm

I'd suggest @jessjwilliamson or @avvvvve whip up something for this and update the musescoreicon font (I don't mind who does it – you both can decide among yourselves :-))

**Comment by oktophonie** [comment]
You can choose what music font to use in the master palette (by default, as of this PR, it'll be whatever music symbols font is currently in force). The font dropdown in Properties is the only means of changing this (which you may want or need to do it you ever change the music font, or for other supremely niche reasons).

The master palette shows SMuFL symbols, so they can exist in any SMuFL font.

**Review by bkunda — APPROVED** [review]

File: src/palette/view/widgets/symboldialog.cpp [hunk 2]
```diff
@@ -82,10 +82,13 @@ SymbolDialog::SymbolDialog(const QString& s, QWidget* parent)
     range = s;          // smufl symbol range
     int idx = 0;
     int currentIndex = 0;
+    Score* score = globalContext()->currentNotation()->elements()->msScore();
+    std::string styleFont = score ? score->style().styleSt(Sid::MusicalSymbolFont).toStdString() : "";
     for (const IEngravingFontPtr& f : engravingFonts()->fonts()) {
         fontList->addItem(QString::fromStdString(f->name()));
-        if (f->name() == "Leland" || f->name() == "Bravura") {
+        if (styleFont != "" && f->name() == styleFont) {
```
Comment (cbjeukendrup) [inline src/palette/view/widgets/symboldialog.cpp:89]:
```suggestion
        if (!styleFont.empty() && f->name() == styleFont) {
```

**Review by cbjeukendrup — APPROVED** [review]
A few minor comments, but generally looks good!

**Pull request merged**